### ISSUE TITLE
Property descriptions missing from Docs Page

### ISSUE THREAD

**Issue by csempesz** [storybookjs/storybook#14876]
**What's the problem?**
Installed SB in a simple Preact project but component property descriptions are not showing up on the Docs page. Not even for the Example components coming with SB. The documentation says: "DocsPage is the zero-config default documentation that all stories"
Do I have to install additional packages?

**Is there documentation on this?**
https://storybook.js.org/docs/react/writing-docs/docs-page

**Additional context**
Again this is not working with the examples provided by SB.

**Comment by csempesz** [comment]
Oh...missed that matrix. I tried so many things to make this work...Thanks. Any insight on when might that happen (ie. support for Preact pure components)? 
Since my Preact components are implemented in typescript I thought that the key piece in this puzzle was the **"react-docgen-typescript"** library and I just need to be able to configure SB properly so that it will be called.  And actually, I could get decent information out of my components, props, default values, descriptions, etc using the library directly.
```
const docgen = require("react-docgen-typescript");
const options = {
  savePropsValueAsString: true
};
// Create a parser with using your typescript config
const tsConfigParser = docgen.withCustomConfig("./tsconfigtest.json", options).parse("./components/greeter/Greeter.tsx");
```
Why can't SB use this library for Preact components as well?

**Comment by shilman** [comment]
@csempesz `react-docgen-typescript-plugin` annotates your components with a `__docgenInfo` field that contains the relevant information. Then there is a snippet of code that translates this into `ArgTypes` which is Storybook's internal data structure. This snippet is different for each framework because the source data format is typically different. If the source format for `preact` is the same as for `react, then it would simply need to run the same snippet of code, which `@storybook/addon-docs` is not currently configured for. Would you be interested in contributing that to Storybook?

**Comment by csempesz** [comment]
Absolutely.  Let me know what you have in your mind. I presume this work will be done for addon-docs?

**Comment by shilman** [comment]
Awesome! 💯 

This is the operative code:

https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/react/config.ts

And here is a hand-wavey guide for what it's doing:

https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/multiframework.md

If you simply copy the `frameworks/react` to `frameworks/preact`, then run:

```
yarn build addon-docs --watch
```

And then in another window run:

```
cd examples/preact-kitchen-sink
yarn storybook
```

That should get you pretty far. You might need to set up Typescript in that repo as well.

There's a contributing guide here (that is getting a rewrite soon): https://github.com/storybookjs/storybook/blob/next/CONTRIBUTING.md#development-guide

And if you want to chat about anything you can jump onto our Discord: https://discord.gg/storybook

**Comment by csempesz** [comment]
Hi Michael,
it's been two months since our communication and I'm sorry I disappeared for that long. I was overwhelmed with work in releasing our next major version of Oracle JET. It was also not sure whether we will embark on the SB boat but now that things have cleared,  I'm back to evaluate again SB especially the Preact support for it. 
I followed your original guidance and created a "preact" entry under addons/docs/src/frameworks, then re-built SB.
I added a new Preact component (Greeter) under examples/preact-kitchen-sink. This web component is a class extending Preact Component so it has a render function that returns jsx (as opposed to the other examples that are functional components). Added a story under examples/preact-kitchen-sink that uses Greeter and then run **yarn storybook**
When the page comes up and I click on the Greeter component I receive 
```
ReferenceError: h is not defined
    at Object.Template (http://localhost:9009/main.iframe.bundle.js:33294:3)
    at finalStoryFn (http://localhost:9009/main.iframe.bundle.js:14299:32)
    at http://localhost:9009/main.iframe.bundle.js:10446:21
```
Questions:
- since the default/base/common preset for Docs is using { pragma: 'React.createElement', pragmaFrag: 'React.Fragment' } when setting up the jsxPlugin...do I need to create a Preact specific preset? 
- How can I debug the docs addon using the sample examples?

**Comment by yoruvo** [comment]
We are currently struggling with this issue in Storybook 6.4, trying both react-docgen and react-docgen-typescript.

@shilman The matrix was since updated to say that Docs are supported for preact, but this doesn't seem to be the case.

Is this issue still open, or has it been addressed since? Because we are getting the same issue as @csempesz.

**Comment by KrofDrakula** [comment]
@yoruvo @csempesz You might like to know that I've just fixed issues with TSX imports; in #15564 the `h` and `Fragment` imports are not required anymore, neither are the `@jsx h` pragmas. You can have a look at the diff and extract the parts you need before this makes its way into the core.

**Comment by shilman** [comment]
We’re cleaning house! Storybook has changed a lot since this issue was created and we don’t know if it’s still valid. Please open a new issue referencing this one if:
- this is still a problem in SB 7.x
- you can provide a consistent reproduction in 7.x (at https://storybook.new/ or per [repro docs](https://storybook.js.org/docs/react/contribute/how-to-reproduce)).
